conan/tools/cmake/presets.py

```diff
@@ -126,23 +145,22 @@ def write_cmake_presets(conanfile, toolchain_file, generator, cache_variables):
     multiconfig = is_multi_configuration(generator)
 
     if os.path.exists(preset_path):
-        # We append the new configuration making sure that we don't overwrite it
         data = json.loads(load(preset_path))
-        if multiconfig:
-            new_build_preset_name = _build_preset_name(conanfile)
-            already_exist = any([b["name"] for b in data["buildPresets"]
-                                 if b["name"] == new_build_preset_name])
-            if not already_exist:
-                data["buildPresets"].append(_add_build_preset(conanfile, multiconfig))
+        build_preset = _build_preset(conanfile, multiconfig)
```

Comment:
The below code was a bug, previously I only appended a preset if it didn't exist, but it might be changed, it is needed to overwrite it, so now I search the index and update.